scripts/calculator.js

```diff
@@ -66,22 +77,163 @@ export class Calculator {
+            case "reciprocal":
+                this.display.append("1/");
+                return;
+            case "!":
+            case "factorial":
+                this.display.append("!");
+                return;
+            case "log":
+                this.display.append("log(");
+                return;
+            case "ln":
+                this.display.append("ln(");
+                return;
+            case "%":
+            case "mod":
+                this.display.append("%");
+                return;
+            case "exp":
+                this.display.append("e^");
+                return;
+            case "abs":
+                this.display.append("abs(");
+                return;
+            case "plusminus":
+                this.sendQuery(`(-1) * (${this.display.get()})`);
+                return;
+            case "sin":
+            case "cos":
+            case "tan":
+            case "asin":
+            case "acos":
+            case "atan":
+            case "floor":
+            case "ceil":
+            case "round":
+            case "cbrt":
+                this.display.append(`${input}(`);
+                return;
+            case "deg":
+            case "rad":
+                this.toggleDegRad();
+                return;
+            case "fn1":
+            case "fn2":
+                this.toggleFnMode();
+                return;
+            case "f-e":
+            case "ex":
+                this.toggleResultMode();
+                return;
+            case "mc":
+                this.memory.clear();
+                return;
+            case "mr":
+                this.display.set(this.memory.recall());
+                return;
+            case "m+":
+                this.memory.plus(this.display.get());
+                return;
+            case "m-":
+                this.memory.minus(this.display.get());
+                return;
+            case "ms":
+                this.memory.store(this.display.get());
                 return;
         }
     }
 
+
+    sendQuery(q) {
+        this.evaluator.postMessage({
+            q,
+            degreeMode: this.degRadBtn.value === "deg",
+            exponentialResult: this.resultModeBtn.value === "ex",
+        });
+    }
+
+
     // Handle message events from evaluator worker.
     handleResult(e) {
         if (e.data.success) {
             this.display.set(e.data.result);
         }
         else {
-            alert(e.data.error);
+            alert(`Error: ${e.data.error.message}`);
             console.error(e.data.error);
         }
     }
 
+    toggleDegRad() {
+        if (this.degRadBtn.value === "deg") {
+            this.degRadBtn.value = "rad";
+            this.degRadBtn.textContent = "RAD";
+            this.degRadBtn.ariaLabel = "Radian Mode";
+        }
+        else {
+            this.degRadBtn.value = "deg";
+            this.degRadBtn.textContent = "DEG";
+            this.degRadBtn.ariaLabel = "Degree Mode";
+        }
```

Comment:
Just a thought, could we make it more robust using ternary? 
Same can be applicable for the `toggleFnMode` and `toggleResultMode`.

```suggestion
       const isDeg = this.degRadBtn.value === "deg";
       this.degRadBtn.value = isDeg ? "rad" : "deg";
       // remaining code
}
```

Reply:
Fixed it.
https://github.com/dhruvdhaduk-simform/scientific-calc/pull/4/commits/7a577e8d8592eb21cd2d78bba66a66e5af6d96f8